src/spaceapi_client.rs

```diff
@@ -1,84 +1,203 @@
 //extern crate spaceapi;
+use std::io::prelude::*;
 use rustc_serialize::json;
 use hyper::Client;
-use std::io::Read;
-use spaceapi::Optional::{Value,Absent};
+use spaceapi::Optional::{self, Value, Absent};
 use spaceapi::{Status};
 use spaceapi::sensors::{PeopleNowPresentSensor};
 
-struct SpaceApiClient {
+use std::time::Duration;
+use time::SteadyTime;
+use std::env;
+use std::fs::{self, File};
+use std::io;
+
+pub struct SpaceApiClient {
   last_fetch: i64,
+  webcams: Vec<String>,
+}
+
+impl SpaceApiClient {
+  pub fn new() -> SpaceApiClient {
+    let mut s = SpaceApiClient{
+      last_fetch: 0,
+      webcams: Vec::new(),
+    };
+    
+    s.fetch_webcams();
```

Comment:
I'm not sure if the constructor should have I/O side effects. Maybe move this to an `init()` method?